js/inner-page.js

```diff
@@ -0,0 +1,43 @@
+import { fetchData } from "./api.js";
+
+const ui = {
+    itemsContainer: document.querySelector('.items-container')
+}
+
+const appendItems = async () => {
+    const items = await fetchItems();
+    ui.itemsContainer.innerHTML = '';
+    checkIfEmpty(items);
+    items.forEach(item => {
+        const newItem = buildNewItem(item);
+        ui.itemsContainer.appendChild(newItem);
+    })
+}
+
+async function fetchItems() {
+    const category = new URLSearchParams(window.location.search).get('category');
+    const items = await fetchData(`items?categoryId=${category}`);
```

Comment:
The `fetchData` function is a bit problematic, it's too generic in general -- I'd advise to have a separate method for each entity:

```
export const fetchCategory = (categoryId) => ...
export const fetchFeaturedProducts = () => ...
```

"Magic" strings, such as the value of the parameter you passed it -- are prone to errors. It's a good idea to relocate these strings so that they would not be mixed with surrounding business logic.